cli/azd/pkg/output/ux/preview_provision.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package ux
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/fatih/color"
+)
+
+// PreviewProvision defines a ux item for displaying a provision preview.
+type PreviewProvision struct {
+	Operations []*Resource
+}
+
+// OperationType defines the valid options for a resource change.
+type OperationType string
```

Comment:
Wondering if we should implement [`fmt.Stringer`](https://pkg.go.dev/fmt#Stringer) on this type, so we could display `ChangeTypeNoChange` as "Unchanged" instead of "NoChange".  This is the one of these that ends up not being an English word, and having the text smushed together like this makes the output feel a little more "computer'y" and less friendly.